src/sage/categories/groups.py

```diff
@@ -451,6 +451,51 @@ def conjugacy_class(self, g):
             from sage.groups.conjugacy_classes import ConjugacyClass
             return ConjugacyClass(self, g)
 
+        def minimum_generating_set(self):
+            """
+            Return a list of the minimum generating set of this group.
+
+            EXAMPLES::
+
+                sage: G = GL(2,GF(3))
+                sage: G.minimum_generating_set()
+                [
+                [1 2]  [1 2]
+                [0 1], [1 1]
+                ]
+
+                sage: G = SymmetricGroup(3)
+                sage: s = G.minimum_generating_set(); s
+                [(2,3), (1,2,3)]
+                sage: s[0].parent()
+                Symmetric group of order 3! as a permutation group
+
+                sage: A5 = AlternatingGroup(5)
+                sage: A5.minimum_generating_set()
+                [(1,2,3,4,5), (3,4,5)]
+
+                sage: H = groups.matrix.Heisenberg(1,3); H
+                Heisenberg group of degree 1 over Ring of integers modulo 3
+                sage: H.minimum_generating_set()
+                [
```

Comment:
I got this as the output instead :
```
[
[1 0 0]  [1 1 0]
[0 1 1]  [0 1 0]
[0 0 1], [0 0 1]
]
```
Although it's the same thing, I think it's good to update it.